### ISSUE TITLE
Convert Javadoc of configuration properties to Asciidoc

### ISSUE THREAD

**Issue by emmanuelbernard** [quarkusio/quarkus#3433]
Replace the Javadoc for `@ConfigItem` and friends to Javadoc format.

Make sure the documentation is useful for people and complete. It will be what the guide will show.

From 

```
        /**
         * Schema genration config
         * The default is {@code none}.
         * <p>
         * Same as JPA's javax.persistence.schema-generation.database.action.
         */
        @ConfigItem
        public Optional<String> generation;
```

Go for 

```
        /**
         * Select whether the database schema is generated or not.
         * Options are `none`, `create`, `drop-and-create`, `drop`, `update`}.
         * The default is `none`.
         *
         * `drop-and-create` is particularly useful in Quarkus Lice Coding mode
         * 
         * Same as JPA's javax.persistence.schema-generation.database.action.
         */
        @ConfigItem
        public Optional<String> generation;
```

All extensions should use that model.

**Comment by dmlloyd** [comment]
No, this is not right.  The JavaDoc ends up on a JavaDoc site.  It's not hard to parse the JavaDoc and then put it in to Asciidoc correctly.  Let's do it right.

**Comment by gsmet** [comment]
This has been implemented and is optional.
